client/components/marketing-survey/cancel-purchase-form/index.jsx

```diff
@@ -678,9 +686,26 @@ class CancelPurchaseForm extends Component {
 			return translate( 'Remove product' );
 		}
 
-		if ( isPlan( purchase ) ) {
-			return translate( 'Cancel plan' );
+		if ( hasAmountAvailableToRefund( purchase ) ) {
+			if ( isDomainRegistration( purchase ) ) {
+				return translate( 'Cancel domain and refund' );
+			}
+			if ( isSubscription( purchase ) ) {
+				return translate( 'Cancel subscription' );
+			}
+			if ( isOneTimePurchase( purchase ) ) {
+				return translate( 'Cancel and refund' );
+			}
 		}
+
+		if ( isDomainRegistration( purchase ) ) {
+			return translate( 'Cancel domain' );
```

Comment:
I feel like getting rid of the "Cancel domain" wording (in cases where we are only turning off auto-renew and the domain actually isn't being removed) was a key goal of the issue also.  On the other hand, I'm not sure we ever got clarity on how to word it.  Simply letting it use "Cancel subscription" like the others would mostly make sense, although I kind of see how it feels weird to say that for a domain :shrug:  Maybe we can ask for advice on this one?

Reply:
One possibility to consider for these cases might be this: "Cancel domain subscription"?